### ISSUE TITLE
Inclusion of Perturbation Targets in Variable Genes for Downstream Analysis in Mixscape

### ISSUE THREAD

**Issue by shahrozeabbas** [satijalab/seurat#3731]
Hello,

I have a two-part question regarding the newer Mixscape tool as it relates to the downstream results based on whether or not perturbation targets are considered variable for analysis. I am approaching the issue from an informatics perspective and would like to hear any thoughts on the biological implications as well. 

For a CRISPRi or a knockout experiment that is combined with scRNA-seq, it is my understanding that a single perturbation _may_ illicit a transcriptome wide effect that should be detectable from sequencing. It could also not happen, but if by design this does happen, then it might make sense that those targets would be considered variable features for downstream. 

The code below uses the `eccite` dataset from the vignette and determines which perturbation targets are also showing up as variable features for Seurat. There are 5 genes: CAV1, CD86, MYC, NFKBIA, and POU2F2 that are considered variable features and are targets for the experiment. 

`# load packages`
`pkgs <- c('Seurat', 'SeuratData', 'dplyr')`
`invisible(sapply(pkgs, require, character.only = TRUE))`

`# load data`
`eccite <- LoadData(ds = "thp1.eccite")`

`# set default assay and scale data`
`DefaultAssay(object = eccite) <- 'RNA'`
`eccite <- NormalizeData(object = eccite) %>% FindVariableFeatures() %>% ScaleData()`

`# store variable gene list`
`var_genes <- VariableFeatures(eccite)`
`targets <- levels(eccite$gene)[-which(levels(eccite$gene) == 'NT')]`

`# find targets that are also variable genes`
`var_targets <- targets[which(targets %in% var_genes)]`

My question is based on default executions of `ScaleData` and `RunPCA`, which only consider variable features. After running `ScaleData`, the Seurat object will populate a `scale.data` table with genes only specified in the `features` parameter or by default all variable features. Subsequently, `RunPCA` will only compute using genes that are present in `scale.data` and do not have 0 variance. If your perturbation targets are not present in the list of variable features, unless specified, `ScaleData` and `RunPCA` do not consider all perturbation targets in the Mixscape analysis of a CRISPRi sequencing experiment. 

Furthermore, of the 5 targets that are variable, I am confused as to why each one has no proportion of KO cells based on the faceted bar plot. So, of the 25 targets, 5 that are variable features and used for scaling and PCA contain little to or no proportion of KO cells, and of the 20 remaining which are not variable genes used in the analysis there are 10 with some proportion of KO cells. 
[mixscape_perturb_cells.pdf](https://github.com/satijalab/seurat/files/5548834/mixscape_perturb_cells.pdf)

In summary, why are not all targets included to be used with `ScaleData` and `RunPCA` in perturbation analysis even if they aren't considered variable enough and why would targets that are variable features contain no proportion of cells with those genes knocked out?

Thank you for your input.

**Comment by satijalab** [comment]
When we run ScaleData and RunPCA, the goal is to identify major sources of variation in the dataset, so that we can match cells that receive targeting gRNA to the most similar cells that receive NT gRNA. These sources of variation are often confounding (replicate, cell cycle, etc.), and the goal of this first step is to remove confounding sources of variation.

In the ensuing steps, we use differential expression to identify genes, not variable features.  If there are no DE genes between CAV1KO and NT cells, then we don't have any evidence for perturbation
